### PR TITLE
Add proper support for custom Swift toolchains

### PR DESCRIPTION
**Pull request by keith**
This adds support for replacing the
`__BAZEL_CUSTOM_XCODE_TOOLCHAIN_PATH__` argument with the absolute path to
the custom toolchain defined by the TOOLCHAINS environment variable
(passed with --action_env). This allows us to fully support custom Swift
toolchains and linking their libraries

https://github.com/bazelbuild/rules_swift/pull/1025

### CODE REVIEW

File: crosstool/wrapped_clang.cc [hunk 1]
```diff
@@ -292,16 +295,54 @@ std::string GetCurrentDirectory() {
...
+std::string GetToolchainPath(const std::string &toolchain_id) {
+  // NOTE: This requires all toolchains to contain a 'clang' executable. This
+  // is true today for custom Swift toolchains, but could change in the future.
+  std::string output = exec("xcrun --find clang --toolchain " + toolchain_id);
+
+  if (output.empty()) {
+    std::cerr << "Error: TOOLCHAINS was set to '" << toolchain_id
+      << "' but no toolchain with that ID was found" << std::endl;
+    exit(EXIT_FAILURE);
+  } else if (output.find("XcodeDefault.xctoolchain") != std::string::npos) {
```
Comment (brentleyjones) [inline crosstool/wrapped_clang.cc:321]: I think we know what DEVELOPER_DIR actually is, and can instead check for that? Not sure if that would actually be better in any way though.

**Review by brentleyjones — APPROVED** [review]

File: crosstool/wrapped_clang.cc (hunk 1 above)
Comment (brentleyjones) [reply crosstool/wrapped_clang.cc:321]: Ignore this. That would be more brittle.

File: crosstool/osx_cc_configure.bzl [hunk 2]
```diff
@@ -56,8 +56,8 @@ def _compile_cc_file(repository_ctx, src_name, out_name):
         "--sdk",
         "macosx",
         "clang",
-        "-mmacosx-version-min=10.13",
-        "-std=c++11",
+        "-mmacosx-version-min=10.15",
+        "-std=c++17",
```
[the remark this answers is not in the document]
Comment (keith) [reply crosstool/osx_cc_configure.bzl:60]: yea, `filesystem` requires both of these

[the remark this answers is not in the document]
Comment (keith) [reply crosstool/osx_cc_configure.bzl:60]: this shouldn't have any actual impact on our minimum supported versions

**Pull request merged**